.github/workflows/docker.yaml

```diff
@@ -0,0 +1,56 @@
+name: Docker Build
+
+concurrency:
+  group: ${{github.repository}}-${{ github.ref_name }}-docker
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+
+# Permissions needed
+# permissions:
+#   contents: read
+#   packages: write
+#   pull-requests: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          # list of Docker images to use as base name for tags
+          images: ghcr.io/${{github.repository}}
```

Comment:
let's expose the image name as input parameter (with default to current repo name), so that:
1. we can reuse this workflow also in repos where there are multiple images associated, e.g.: https://github.com/zaphiro-technologies/data-download-backend/blob/main/.github/workflows/docker.yaml
1. we can use a different name for the docker image as needed.